backend/TraderX/src/main/java/cmpe451/group6/rest/transaction/service/TransactionService.java

```diff
@@ -0,0 +1,204 @@
+import java.util.Date;
+
+@Service
+public class TransactionService {
+
+    @Autowired
+    private TransactionRepository TransactionRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private FollowRepository followRepository;
+
+    @Autowired
+    private EquipmentRepository equipmentRepository;
+
+    @Autowired
+    private AssetRepository assetRepository;
+
+    @Autowired
+    private ModelMapper modelMapper;
+
+    public List<Transaction> getTransactionsByUser(String username, String requesterName) {
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            throw new CustomException("There is no user named " + username + ".", HttpStatus.NOT_ACCEPTABLE);
+        } else if (user.getIsPrivate() && (followRepository.getFollowDAO(requesterName, username) == null
+                || followRepository.getFollowDAO(requesterName, username).getFollowStatus() != FollowStatus.APPROVED)) {
+            throw new CustomException("The requested user's profile is private and requester is not following!",
+                    HttpStatus.NOT_ACCEPTABLE);
+        } else {
+            List<Transaction> transactions = TransactionRepository.findByUser_username(username);
+            return transactions;
+        }
+    }
+
+    public List<TransactionDTO> getTransactions() {
+        List<TransactionDTO> transactionDTOs = new ArrayList<TransactionDTO>();
+        TransactionRepository.findAll()
+                .forEach(item -> transactionDTOs.add(modelMapper.map(item, TransactionDTO.class)));
+        return transactionDTOs;
+    }
+
+    public List<TransactionDTO> getTransactionsByCode(String code) {
+        List<TransactionDTO> transactionDTOs = new ArrayList<TransactionDTO>();
+        TransactionRepository.findByEquipment_code(code)
+                .forEach(item -> transactionDTOs.add(modelMapper.map(item, TransactionDTO.class)));
+        return transactionDTOs;
+    }
+
+    public List<TransactionDTO> getTransactionByDateBetween(Date start, Date end) {
+        List<TransactionDTO> transactionDTOs = new ArrayList<TransactionDTO>();
+        TransactionRepository.findByDateBetween(start, end)
+                .forEach(item -> transactionDTOs.add(modelMapper.map(item, TransactionDTO.class)));
+        return transactionDTOs;
+    }
+
+    public int numberOfTransactions() {
+        return TransactionRepository.countAll();
+    }
+
+    public int numberOfTransactionByUser(String username, String requesterName) {
+        User user = userRepository.findByUsername(username);
+        if (user == null) {
+            throw new CustomException("There is no user named " + username + ".", HttpStatus.NOT_ACCEPTABLE);
+        } else if (user.getIsPrivate() && (followRepository.getFollowDAO(requesterName, username) == null
+                || followRepository.getFollowDAO(requesterName, username).getFollowStatus() != FollowStatus.APPROVED)) {
```

Comment:
followService#isPermitted does the same job as this huge if does. Better we changed this.